### DISCUSSION TITLE
Welcome to chatgpt4whatsapp Discussions!

### DISCUSSION THREAD

**Discussion by clovelee** [chatgpt4social/chatgpt4whatsapp#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by superhandishere** [comment]
is this chatbot working only when the browser is online? while I have installed extension on chrome and set according to guidance, will the chatbot automatically appear? I'm jutst making sure the one I dealt with is correct

**Comment by clovelee** [comment]
Firstly, to use the plugin, you must keep the WhatsApp web page logged in and open all the time so that the plugin can function properly. Secondly, if you are using an API as your AI source, you can use it directly. However, if you are using another web-based AI source, you will need to open the web page and log in once. Once you have logged in successfully, you can close the AI source web page. For example, if you are using the chat.openai.com web page as your AI source, you need to log in once and then you can start using it.

> 2023年4月12日 00:08，superhandishere ***@***.***> 写道：
> 
> is this chatbot working only when the browser is online? while I have installed extension on chrome and set according to guidance, will the chatbot automatically appear? I'm jutst making sure the one I dealt with is correct

**Reply by superhandishere** [reply]
Thanks for your reply, I do use an api but what do you mean by using another web-based AI source?

**Reply by maooson** [reply]
Chrome store version is out of date and there is a bug in the API, please download the latest version: https://aow.me/chatgpt4whatsapp_download

**Comment by MikeBivins** [comment]
Hi, I’m using it for some basic chit-chat right now and I have some feedback to give.

When I use ChatGPT through the webpage instead of the API, I can run all models except GPT4. Whenever I select GPT4, it always gives me a 403 error. I think my access token is set up correctly because the other Davinci models work fine. But the API works as expected.
Also, when I use the ChatGPT webpage, I’ve noticed that in an ongoing dialogue, it only edits the first message and adds the latest one to it (including the default prompt). This makes it impossible to have a fluent dialogue and it’s much worse than using regular ChatGPT.
It would be nice if ChatGPT could be triggered by replying to its last message without using (@gpt) and also include that message as a parent message for context reasons.
Other than that, I think it’s a great project. It replies almost instantly. How fast is the reply speed with Bing?

**Reply by maooson** [reply]
1. Keep the model consistent. If you are using GPT4, you need to ensure that the model selected on the ChatGPT web page is consistent with the model in the Chrome extension(ChatGPT for WhatsApp), both are GPT4, and the model on the extension(ChatGPT for WhatsApp) needs to be saved to take effect after being changed.
2. Maybe it's a bug 😊
3. There is a `Trigger Mode` config item in setting page. Set to `Always` mode is ok.

**Reply by MikeBivins** [reply]
sadly it still doesnt utilize a context meaning every single message sent is a new conversation making it impossible to e.g. reply to something gpt says or maintain a dialogue its basically a gpt inference once thats it no further interaction possible.

I also noticed that you have removed the options to add api keys from somewhere other than openai why is that

**Comment by maooson** [comment]
> sadly it still doesnt utilize a context meaning every single message sent is a new conversation making it impossible to e.g. reply to something gpt says or maintain a dialogue its basically a gpt inference once thats it no further interaction possible.
> 
> I also noticed that you have removed the options to add api keys from somewhere other than openai why is that
1. The context is always there. If you use the openai api, the context only retains the context of 7 conversations. hard code.
![image](https://github.com/chatgpt4social/chatgpt4whatsapp/assets/1394747/14172ccc-e48d-4c43-8dea-6d07f2b22680)
3. This is a configurable feature, I can turn this option on if you find it useful.

**Reply by MikeBivins** [reply]
Excellent support. to 2. I have been sporadically using it, and if there are no negative consequences, I would be thrilled to keep it activated.

3. I understand that by utilizing the Webgpt/chatgptapi instead of the official API, there is a possibility that openai may impose restrictions or ban the account. Has anyone ever experienced such consequences solely by using chatgpt as an API?

**Reply by maooson** [reply]
2. I promise that the extension with no malicious intent. If you find it useful, feel free to continue using it.
3. The ChatGPT for WhatsApp extension operates with the same effectiveness as using the service directly on the webpage. There's no risk of your account being flagged or banned due to its usage. Furthermore, the OpenAI API key is stored locally within your browser, ensuring the safety and confidentiality of your API key without any risk of exposure.

**Comment by MikeBivins** [comment]
Great job, thank you once again. Now sometimes Bard seems to have context, possibly due to Google combining the messages. However, it has improved. I don't understand why the /gpt is behind paywall but oke @gpt works well. The extension is currently excellent, and I recommend it to others privately all the time. 
By any chance, is there a way to test the paid option with Bing for a day or two? 

edit: If I am located in a country where Bard still uses Palm-2 instead of GEMINI, can I bypass this by using a VPN and then reconnect to Bard? 
2. Is there a method to make it respond in one continuous conversation in Bard instead of creating a new one? Sometimes it does this after a certain amount of time has passed, I believe. However, if it's possible to always send my query using the last conversation, it would greatly help in maintaining consistent context.
